sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/PruneFileSourcePartitionsSuite.scala

```diff
@@ -109,9 +110,26 @@ class PruneFileSourcePartitionsSuite extends PrunePartitionSuiteBase {
     }
   }
 
+  test("SPARK-35985 push filters for empty read schema") {
```

Comment:
nit: perhaps we should add SPARK-36776 here too?

Reply:
Added. Thanks

Reply:
I think we don't need a new JIRA here as this is a backport case.

Reply:
yea make sense, we can just change the PR title